dtfs-central-api/src/cron-scheduler-jobs/create-utilisation-reports/index.ts

```diff
@@ -0,0 +1,75 @@
+import { asString, CronSchedulerJob, getCurrentReportPeriodForBankSchedule } from '@ukef/dtfs2-common';
+import { getOneUtilisationReportDetailsByBankId, saveNotReceivedUtilisationReport } from '../../services/repositories/utilisation-reports-repo';
+import { getAllBanks } from '../../services/repositories/banks-repo';
+import { Bank } from '../../types/db-models/banks';
+import { validateUtilisationReportPeriodSchedule } from './utilisation-report-period-schedule-validator';
+
+const { UTILISATION_REPORT_CREATION_FOR_BANKS_SCHEDULE } = process.env;
+
+/**
+ * Checks if the current bank report is missing
+ * @param bank - The bank
+ * @returns Whether or not the bank report is missing
+ */
+const isCurrentBankReportMissing = async (bank: Bank): Promise<boolean> => {
+  const currentReportPeriod = getCurrentReportPeriodForBankSchedule(bank.utilisationReportPeriodSchedule);
+  const currentUtilisationReportForBank = await getOneUtilisationReportDetailsByBankId(bank.id, { reportPeriod: currentReportPeriod });
+  return !currentUtilisationReportForBank;
+};
+
+/**
+ * Checks if the current bank has a valid utilisation report period schedule
+ * @param bank - The bank
+ * @returns Whether or not the bank utilisation report period schedule is valid
+ */
+const isValidUtilisationReportPeriodScheduleOnBank = (bank: Bank): boolean => {
+  try {
+    validateUtilisationReportPeriodSchedule(bank.utilisationReportPeriodSchedule);
+    return true;
+  } catch (validationError) {
+    console.info('Invalid utilisation report period schedule for bank with id %s. %s', bank.id, validationError);
```

Comment:
Having this as a console.warn triggered an eslint no console warning so I had to reduce it to a info to be able to commit...